src/main/java/org/folio/rest/impl/PermissionUtils.java

```diff
@@ -0,0 +1,56 @@
+package org.folio.rest.impl;
+
+import org.folio.rest.jaxrs.model.OkapiPermission;
+import org.folio.rest.jaxrs.model.Permission;
+
+public class PermissionUtils {
+
+  private PermissionUtils() {
+    
+  }
+  
+  /**
+   * Compare two objects for equality, w/ checks for null, etc.
+   * 
+   * @param a LHS of comparison
+   * @param b RHS of comparison
+   * @return true if equal, false otherwise
+   */
+  public static boolean equals(Object a, Object b) {
+    if ((a == null && b != null) || (a != null && b == null)) {
+      return false;
+    }
+    return (a == null && b == null) || (a.equals(b));
+  }
+
+  /**
+   * Compare an OkapiPermission to a Permission, including definedBy.
+   * 
+   * @param okapiPerm a permission passed from Okapi
+   * @param moduleName the module name to compare against permission's definedBy.moduleName property
+   * @param perm the Permission to compare against
+   * @return true if equal, false otherwise
+   */
+  public static boolean comparePerms(OkapiPermission okapiPerm, String moduleName, Permission perm) {
+    return comparePerms(okapiPerm, perm)
+        && perm.getDefinedBy() != null 
+        && equals(moduleName, perm.getDefinedBy().getModuleName());
```

Comment:
```suggestion
    String otherModuleName = perm.getDefinedBy() == null ? null : perm.getDefinedBy().getModuleName();
    return comparePerms(okapiPerm, perm)
        && Objects.equals(moduleName, otherModuleName);
```
The method should return true if perm.getDefinedBy() == null and moduleName == null.